utils/controller_debug.py

```diff
@@ -891,37 +895,50 @@ def DbgPrint(S="", end="\n"):
         S += "0x%02x " % b
     DbgPrint(S)
 
-    ser.write(bytearray(buff))
-    if timeout != None:
-        DbgPrint("Setting timeout to %.1f" % timeout)
-        oldto = ser.timeout
-        ser.timeout = timeout
-
-    rsp = GetResp(DbgPrint)
-    if timeout != None:
-        ser.timeout = oldto
-
-    if len(rsp) < 3:
-        raise Error("Invalid response, too short")
-
-    crc = crc16.calc(rsp[:-2])
-    rcrc = Build16(rsp[-2:])[0]
-    if crc != rcrc:
-        raise Error(
-            "CRC error on response, calculated 0x%04x received 0x%04x" % (crc, rcrc)
-        )
-
-    if rsp[0]:
-        raise Error("Error %d (0x%02x)" % (rsp[0], rsp[0]))
-        return []
-    return rsp[1:-2]
+    cmdLock.acquire()
+    try:
```

Comment:
These try blocks should be replaced with simply:
```
with cmdLock:
  ...
```
which does acquire/release for you including in case of exceptions